.ci/verify

```diff
@@ -1,19 +1,55 @@
-#!/bin/bash
+#!/bin/bash -x
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
+concourse_cache_dir="$PWD/.cache_concourse"
+use_cache=no
+[ -d "$concourse_cache_dir" ] && use_cache=yes
+
 cd "$(dirname $0)/.."
 
 git config --global user.email "gardener@sap.com"
 git config --global user.name "Gardener CI/CD"
 
+# TODO: use pre-built image instead of installing in every pipeline run
 apt-get update
 apt-get install -y unzip jq parallel
 
 mkdir -p /go/src/github.com/gardener/gardener
 cp -r . /go/src/github.com/gardener/gardener
 cd /go/src/github.com/gardener/gardener
 
-make verify-extended
+if [ $use_cache = yes ] ; then
+  make install-requirements
+
+  # list of all to-be-cached directories
+  cache_dirs=(
+    "$(go env GOCACHE)"
+    "$(golangci-lint cache status | grep -i 'dir:' | awk '{print $2}')"
+    # TODO: cache tools (golangci-lint, mockgen, envtest bins, promtool, protoc, ...)
+  )
+
+  echo "> retrieving all cached dirs from concourse cache dir"
+  for cache in "${cache_dirs[@]}" ; do
+    mkdir -p "$cache"
+    cache_name=$(echo "$cache" | tr ' /' '_' | cut -c 2-)
+    if [ -d "$concourse_cache_dir/$cache_name" ] ; then
+      rm -rf "$cache"
+      mv "$concourse_cache_dir/$cache_name" "$cache"
+    fi
+  done
+
+  # run test instead of test-cov to speed-up PR verification as `-cover` flag is not "cacheable"
+  make check-generate verify
```

Comment:
IIUC, this will not run the `test-cov` rule even for the release job. I guess, at least for releases, we should provide coverage results. Can you change this?

Reply:
Nope, currently the cache is only configured for the master PR job. All other jobs will still run the uncached version (including `test-cov`)

Reply:
OK, is this some configuration in the meta/ci ref? What about the head-update job of the master branch?

Reply:
Yes, exactly. It is configured in `meta/ci` for the PR job (so not only for the master PR job, but for all PRs I guess).
The head-update job doesn't use the cache so far.

Reply:
ref https://github.com/gardener/gardener/commit/d15f35e9ef2775f715a24bd6651f42397988f5d0

Reply:
Thanks, I guess you plan to backport this PR to prevent complexities in the branch.cfg?

Reply:
No, I haven't thought about it so far. I think we don't need extra configuration, as the cache will simply be ignored on older `release-*` branches, which shouldn't hurt.
The cache is pipeline-task-specific, IIUC, so it will stay empty for those old branches.